apps/heft/src/plugins/TypeScriptPlugin/LinterBase.ts

```diff
@@ -123,8 +123,15 @@ export abstract class LinterBase<TLintResult> {
         continue;
       }
 
-      const version: string = sourceFile.version;
-      if (cachedNoFailureFileVersions.get(filePath) !== version || options.changedFiles.has(sourceFile)) {
+      // Older compilers don't compute the ts.SourceFile.version.  If it is missing, then we can't skip processing
```

Comment:
In that case, should we use `crypto` to manually compute a hash?

Reply:
I thought about this, but it might be expensive, maybe as expensive as re-linting the file. As I recall `package-deps-hash` relies on Git hashing specifically because it was significantly faster than trying to compute a hash in JavaScript.

Frankly I'm not super concerned about special case performance optimizations for outdated versions of the compiler, unless someone measures this to be significant in some real world repo.